deepchem/data/data_loader.py

```diff
@@ -1689,17 +1690,16 @@ def create_dataset(self,
         """
 
         def shard_generator():
-            entries = self._get_shards(inputs)
-            for i, shard in enumerate(entries):
-                X = np.array(self._featurize_shard(shard))
-                y = X[0].get_true_val()
-                w = np.array([X[0].get_weight()])
-                ids = np.array([i])
+            for shard_num, shard in enumerate(
+                    self._get_shards(inputs, shard_size)):
+                X, y = self._featurize_shard(shard)
+                w = np.array([x.get_weight() for x in X])
+                ids = np.array([shard_num])
```

Comment:
This is not correct; ids and w should be the same shape and all datapoints in the shard should not have different ids. Each datapoint in the dataset should have a unique id

Reply:
Can you add a test to check the shape of ids?